### PR TITLE
Revert: Don't generate CSharpStatementIRNode for whitespace

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Reverting https://github.com/aspnet/Razor/pull/1127

This is not a full revert. This is still not generating line pragmas for whitespace nodes.

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs [hunk 1]
```diff
@@ -77,7 +77,13 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
-            if (node.Source != null)
+            if (node.Source != null && string.IsNullOrWhiteSpace(node.Content))
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]:
```C#
if (string.IsNullOrWhiteSpace(node.Content))
{
    // We need to generate line mappings for whitespace to provide intellisense.
    Context.AddLineMappingFor(node);
    Context.Writer.WriteLine(node.Content);
}
else
{
...
}
```

`AddLineMapping` does nothing if the source is `null`.

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CodeBlockWithTextElement_DesignTime.mappings.txt [hunk 2]
```diff
@@ -17,3 +17,9 @@ Source Location: (69:2,29 [3] TestFiles/IntegrationTests/CodeGenerationIntegrati
 Generated Location: (1050:28,38 [3] )
 |a+b|
 
+Source Location: (80:2,40 [2] TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CodeBlockWithTextElement.cshtml)
+|
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CodeBlockWithTextElement_DesignTime.mappings.txt]: What happened here?

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers_DesignTime.mappings.txt [hunk 3]
```diff
@@ -48,172 +48,186 @@ Source Location: (463:15,63 [4] TestFiles/IntegrationTests/CodeGenerationIntegra
...
+
 Source Location: (507:16,31 [30] TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers.cshtml)
 |true ? "checkbox" : "anything"|
-Generated Location: (3221:70,31 [30] )
+Generated Location: (3253:72,31 [30] )
 |true ? "checkbox" : "anything"|
 
+Source Location: (542:16,66 [18] TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers.cshtml)
+|
+                |
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers_DesignTime.mappings.txt]: The whitespace bits don't seem right.

File: src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs (hunk 1 above)
Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: True. But if we are not going to have a line mapping, what is the point of writing the whitespace content?

Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Super solid point. With how the code is currently you have the same problem though.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs]: Oh lol 🙈

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CodeBlockWithTextElement_DesignTime.mappings.txt (hunk 2 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/CodeBlockWithTextElement_DesignTime.mappings.txt]: It's the padding.

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers_DesignTime.mappings.txt (hunk 3 above)
Comment (ajaybhargavb) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/CodeGenerationIntegrationTest/ComplexTagHelpers_DesignTime.mappings.txt]: It's the padding.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**